android-components/build.gradle

```diff
@@ -63,6 +63,27 @@ allprojects {
             name "Mozilla"
             url "https://maven.mozilla.org/maven2"
         }
+
+        if (ExtraRepositories.mozillaStaging) {
+            maven {
+                name "Mozilla Staging"
+                url "https://maven-default.stage.mozaws.net/maven2"
+            }
+        }
+
+        if (ExtraRepositories.mozillaNightly) {
+            maven {
+                name "Mozilla Nightly"
+                url "https://nightly.maven.mozilla.org/maven2"
+            }
+        }
+
+        if (ExtraRepositories.mozillaNightlyStaging) {
+            maven {
+                name "Mozilla Nightly Staging"
+                url "https://maven-nightly.stage.mozaws.net/maven2"
+            }
+        }
```

Comment:
Nightlies should probably live in a separate location from the other maven packages.  The current proposal is to:
  - Use the `nightly` maven repository for nightly releases
  - Use the `release` maven repository for the actual releases
  - Use the `nightly-staging` and `staging` repositories for preview releases built from PRs.
  - Use groupId  `org.mozilla.appservices` for actual releases
  - Use groupId `org.mozilla.appservices.nightly` for nightly/preview releases.

However, this is very flexible and we can follow whatever the Android team thinks is best.